github/Team.py

```diff
@@ -249,6 +257,37 @@ def set_repo_permission(self, repo, permission):
             "PUT", self.url + "/repos/" + repo._identity, input=put_parameters
         )
 
+    def update_team_repository(self, repo, permission):
+        """
+        :calls: `PUT /orgs/:org/teams/:team_slug/repos/:owner/:repo <https://developer.github.com/v3/teams/#add-or-update-team-repository>`_
+        :param repo: string or :class:`github.Repository.Repository`
+        :param permission: string
+        :rtype: bool
+        """
+        # TODO Add support for Content-Length: 0 when no parameters are added
+        # Original note from the docs:
+        # Note that, if you choose not to pass any parameters, you'll need to set Content-Length to zero when calling out to this endpoint. For more information, see "HTTP verbs."
+
+        assert repo is isinstance(repo, github.Repository.Repository) or isinstance(
+            repo, str
+        ), repo
+        repo_url_param = repo
+        if repo is github.Repository.Repository:
```

Comment:
``if isinstance(repo, github.Repository.Repository):``

Reply:
Thanks, changed in https://github.com/PyGithub/PyGithub/pull/1509/commits/e51fa257bd254e506cfd6c1f93219c8f802f84ba